### PR TITLE
heat flux densities

### PR DESCRIPTION
**Pull request by tjhei**
Add heat flux densities.

(based on #1326, so let us merge that first and I can rebase)

### CODE REVIEW

**Comment by bangerth** [comment]
Err, that's not the Nusselt number :-) What you compute is the heat flux density. The Nusselt number is something different: https://en.wikipedia.org/wiki/Nusselt_number

**Comment by tjhei** [comment]
Yep, sadly. I think we have the following options:
a) Remove and live with the fact that blankenbach benchmarks don't output the common statistics (but that you have to scale the top heat flux)
b) Write a postprocessor just for blankenbach benchmarks
c) Just call this output heat flux density.
d) Write a postprocessor that actually computes Nusselt no.

I guess c) or a) sounds better than the other options.

**Comment by bangerth** [comment]
I almost more for keeping each postprocessor simple. This would call for just cloning it and creating a second one that computes the average (called "average heat flux density"). This obviates any need for input parameters.

For Blankenbach, I see nothing wrong with having to compute a quantity from one of the numbers output by ASPECT. It's easy enough to do in a spreadsheet, or gnuplot.

**Comment by tjhei** [comment]
> This would call for just cloning it and creating a second one that computes the average

I wanted to avoid copying 100+ lines of code, but if you prefer that, sure.

**Comment by tjhei** [comment]
I did another iteration of this.

File: source/postprocess/heat_flux_densities.cc [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                                  "the (conductive) heat flux density for each boundary "
+                                  "id. The heat flux density is computed in outward "
+                                  "direction, i.e., from the domain to the outside, using the "
+                                  "formula $\frac{1}{|\\Gamma_i|} \\int_{\\Gamma_i} -k \\nabla T \\cdot \\mathbf n$ "
+                                  "where $\\Gamma_i$ is the part of the boundary with indicator $i$, "
+                                  "$k$ is the thermal conductivity as reported by the material model, "
+                                  "$T$ is the temperature, and $\\mathbf n$ is the outward normal. "
+                                  "Note that the quantity so computed does not include any energy "
+                                  "transported across the boundary by material transport in cases "
+                                  "where $\\mathbf u \\cdot \\mathbf n \\neq 0$.")
```
Comment (bangerth) [inline source/postprocess/heat_flux_densities.cc]: Can you add something like this here?
```
Note that the `heat flux' postprocessor computes the same quantity as the one here, but not divided by the area. 
In other words, it computes the \\textit{total} heat flux through a boundary.
```
(And maybe the same text with obvious change in the other postprocessor's file.)

**Comment by bangerth** [comment]
Can you rebase this onto master now that #1326 is in?

**Comment by bangerth** [comment]
OK to merge yourself.

**Pull request merged**